features/step_definitions/opportunities_steps.rb

```diff
@@ -37,20 +40,41 @@
   lots_found.each { |x| x.text.should == lot }
 end
 
+Then (/^I see all the opportunities on the page are in the '(.*)' location/) do |lot|
+  locations_found = all(
+    :xpath,
+    '//*[@class="search-result"]//*[@class="search-result-important-metadata"][1]//*[@class="search-result-metadata-item"][2]'
+  )
+  locations_found.each { |x| x.text.should == lot }
+end
+
+Then (/^I see all the opportunities on the page are for the '(.*)' role/) do |lot|
+  locations_found = all(
+    :xpath,
+    '//*[@class="search-result"]//*[@class="search-result-metadata"][1]//*[@class="search-result-metadata-item"][2]'
+  )
+  locations_found.each { |x| x.text.should == lot }
+end
+
 Then (/^I see all the opportunities on the page are of the '(.*)' status$/) do |status|
   published_or_closed = all(
     :xpath,
     '//*[@class="search-result"]//*[@class="search-result-metadata"][2]//*[@class="search-result-metadata-item"][1]'
   )
   published_or_closed.each do |x|
-    if status == 'Closed'
-      x.text.should == 'Closed'
+    if ['Closed', 'Unsuccessful', 'Cancelled'].include? status
```

Comment:
Not a functional difference, but could remove `Unsuccessful` and `Cancelled`

Reply:
Nah I'll leave it.